hs_core/search_indexes.py

```diff
@@ -51,49 +51,71 @@ class BaseResourceIndex(indexes.SearchIndex, indexes.Indexable):
     editors_names = indexes.MultiValueField(faceted=True)
     editors_count = indexes.IntegerField(faceted=True)
 
-
     def get_model(self):
         return BaseResource
 
     def index_queryset(self, using=None):
-        return self.get_model().objects.filter(Q(raccess__discoverable=True) | Q(raccess__public=True))
+        return self.get_model().objects.filter(Q(raccess__discoverable=True) |
+                                               Q(raccess__public=True))
 
     def prepare_title(self, obj):
-        if hasattr(obj, 'metadata'):
+        if hasattr(obj, 'metadata') and obj.metadata.title.value is not None:
             return obj.metadata.title.value
         else:
             return 'none'
 
+    def prepare_abstract(self, obj):
+        if hasattr(obj, 'metadata') and obj.metadata.description.abstract is not None:
+            return obj.metadata.description.abstract
+        else:
+            return 'none'
+
+    def prepare_author(self, obj):
+        if hasattr(obj, 'metadata'):
+            first_creator = obj.metadata.creators.filter(order=1).first()
+            if first_creator.name is not None: 
```

Comment:
This is a search engine field. You might note that both name and organization are already listed in the "creators" clause below. This clause is redunant and could be eliminated entirely.